internal/runners/packages/import.go

```diff
@@ -144,12 +135,51 @@ func (i *Import) Run(params *ImportRunParams) error {
 		return locale.WrapError(err, "err_commit_changeset", "Could not commit import changes")
 	}
 
-	if err := localcommit.Set(i.proj.Dir(), commitID.String()); err != nil {
+	pg.Stop(locale.T("progress_success"))
+	pg = nil
+
+	// Solve the runtime.
+	rt, rtCommit, err := runtime.Solve(auth, out, i.prime.Analytics(), proj, &commitID, target.TriggerImport, i.prime.SvcModel(), i.prime.Config(), runtime.OptNone)
+	if err != nil {
+		return errs.Wrap(err, "Could not solve runtime")
+	}
+
+	// Output change summary.
+	previousCommit, err := bp.FetchCommit(latestCommit, proj.Owner(), proj.Name(), nil)
+	if err != nil {
+		return errs.Wrap(err, "Failed to fetch build result for previous commit")
+	}
+	oldBuildPlan := previousCommit.BuildPlan()
+	out.Notice("") // blank line
+	dependencies.OutputChangeSummary(out, rtCommit.BuildPlan(), oldBuildPlan)
+
+	// Report CVEs.
+	if err := cves.NewCveReport(i.prime).Report(rtCommit.BuildPlan(), oldBuildPlan); err != nil {
+		return errs.Wrap(err, "Could not report CVEs")
+	}
+
+	// Update the runtime.
+	if !i.prime.Config().GetBool(constants.AsyncRuntimeConfig) {
+		out.Notice("")
+
+		// refresh or install runtime
+		err = runtime.UpdateByReference(rt, rtCommit, auth, proj, out, runtime.OptNone)
+		if err != nil {
+			if !runbits.IsBuildError(err) {
+				// If the error is not a build error we want to retain the changes
+				if err2 := localcommit.Set(proj.Dir(), commitID.String()); err2 != nil {
+					return errs.Pack(err, locale.WrapError(err2, "err_package_update_commit_id"))
+				}
+			}
+			return errs.Wrap(err, "Failed to refresh runtime")
+		}
+	}
+
+	if err := localcommit.Set(proj.Dir(), commitID.String()); err != nil {
```

Comment:
This should happen before the runtime update, otherwise runtime update failing will cause the import to have no effect. This makes sense for `state install` because you're only touching one thing, but for import I think you'd want to retain the broken state so you can fix it.

Reply:
Huh, that's the opposite of what I'd expect. If I do `state import` and my requirements.txt is causing a build failure, I'd want to edit my requirements.txt and try again, not do a bunch of `state uninstall`, `state install` operations to try and fix things.

That said, I'll defer to you, but I wanted to double-check this is what you want.

Reply:
Going a step further, it's like `apt install <long list of deps>` and committing even if there's a dependency failure, and then expecting you to `apt uninstall` and `apt install` your way back to a working system. Now I understand we have `state reset` which may sidestep this issue entirely, but it's still an extra step. With a big red 'x' saying my runtime failed to build, I think it's reasonable as a user to assume nothing was committed.

Reply:
@mitchell-as that's a good point, I'm honestly not sure. I could see an argument for both sides, because maybe the nature of the issue is not one that can be addressed by editing the requirements.txt, and further tweaking is required. In this case you'd need the interim state.

That said, the story didn't ask for a change to this mechanic, so I don't think we should be introducing it here. Especially when the value prop isn't black and white.